framework/include/restart/DataIO.h

```diff
@@ -123,6 +123,13 @@ template<typename T>
 inline void
 dataStore(std::ostream & stream, T & v, void * /*context*/)
 {
+  #ifdef LIBMESH_HAVE_CXX11_TYPE_TRAITS
+    static_assert( std::is_polymorphic<T>::value == false,
+      "Cannot serialize a class that has virtual members!\nWrite a custom dataStore() template specialization!\n\n");
+    // static_assert( std::is_trivially_copyable<T>::value,
+    //   "Cannot serialize a class that is not trivially copyable!\nWrite a custom dataStore() template specialization!\n\n");
```

Comment:
This should work for POD, but it seems like the check is _too strict_. It fails on some `std::pair` types and on `Point` (which technically behave like POD). I'm working on something...